HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,105 @@
+from pyspark import SparkContext
+from pyspark.accumulators import AccumulatorParam
+import sys
+
+# Flags
+UNVISITED = 0
+VISITING = 1
+VISITED = 2
+
+# Class for accumulator
+class BFSAccumulator(AccumulatorParam):
+	def zero(self, val):
+		return {}
+
+	# Get union of keys and set value as min distance from source node
+	def addInPlace(self, accum1, accum2):
+		return { k: min(i for i in (accum1.get(k), accum2.get(k)) if i != None) for k in accum1.viewkeys() | accum2}
```

Comment:
This is probably brilliant, but pretty hard for me to pick apart what is going on.  I'd be willing to venture that you didn't learn to code in an online community college class like I did!